api/server/server.go

```diff
@@ -1642,6 +1643,17 @@ func ServeApi(job *engine.Job) engine.Status {
 	return engine.StatusOK
 }
 
+func removeUnixSocket(sock string) {
+	parts := strings.SplitN(sock, "://", 2)
```

Comment:
Why not use unlink function when unix sock is created. 
use unlink, even if docker daemon crash, killed, the unix sock will be removed after use account equal to zero.